Mathlib/Data/Nat/Digits.lean

```diff
@@ -516,6 +517,59 @@ lemma self_div_pow_eq_ofDigits_drop (i n : ℕ) (h : 2 ≤ p):
     (fun l hl ↦ digits_lt_base h hl)
   exact (ofDigits_digits p n).symm
 
+open BigOperators
+
+theorem sub_one_mul_sum_div_pow_eq_sub_sum_digits' (L : List ℕ) (w₁ : ∀ (l : ℕ), l ∈ L → l < p)
+    (w₂ : ∀ (h : L ≠ []), L.getLast h ≠ 0) : (p - 1) * ∑ i in Finset.range L.length,
+    (ofDigits p L) / p ^ i.succ = (ofDigits p L) - L.sum := by
```

Comment:
I think a slightly better statement (which depends on the `open Finset` above) is:
```suggestion
theorem sub_one_mul_sum_div_pow_eq_sub_sum_digits
    (L : List ℕ) {h} (h_ne_zero : L.getLast h ≠ 0) (h_lt : ∀ l ∈ L, l < p) :
    (p - 1) * ∑ i in range L.length, (ofDigits p L) / p ^ i.succ = (ofDigits p L) - L.sum := by
```

Reply:
sure, why not.
I had taken the formulation from 
https://github.com/leanprover-community/mathlib4/blob/f2f99ebf8c9a9441e79cd57418dc81a9777c217d/Mathlib/Data/Nat/Digits.lean#L227-L228